### PR TITLE
[Mellanox] Add  a new Mellanox platform x86_64-mlnx_msn4700 and new SKU ACS-MSN4700

### PR DESCRIPTION
**Pull request by keboliu**
**- What I did**
Add a new Mellanox platform x86_64-mlnx_msn4700 and new SKU ACS-MSN4700.

**- How I did it**
Add related files to the device folder:

- buffer config templates
- pg lookup profile
- port_config.ini
- sai profile
- sensor conf
- plugins
- and a dummy syseeprom hex for simx platform

**- How to verify it**
run test on the Mellanox simx platform.

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by keboliu** [comment]
retest vs  please

**Comment by keboliu** [comment]
retest please

**Comment by keboliu** [comment]
retest this please

**Comment by jleveque** [comment]
Retest vsimage please

**Comment by keboliu** [comment]
Retest vsimage please

File: device/mellanox/x86_64-mlnx_msn4700-r0/ACS-MSN4700/port_config.ini [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# name          lanes                               alias
+Ethernet0       0,1,2,3,4,5,6,7                     etp1
```
Comment (liat-grozovik) [inline device/mellanox/x86_64-mlnx_msn4700-r0/ACS-MSN4700/port_config.ini]: you should be able to split to 8x50 meaning that your interfaces should be Ethernet0, Ethernet8, etc.

File: device/mellanox/x86_64-mlnx_msn4700-r0/plugins/psuutil.py [hunk 2]
```diff
@@ -0,0 +1 @@
+../../x86_64-mlnx_msn2700-r0/plugins/psuutil.py
```
Comment (liat-grozovik) [inline device/mellanox/x86_64-mlnx_msn4700-r0/plugins/psuutil.py]: need to verify if the layout is exactly as 2700. if not, create a new file dedicated for this system.

File: device/mellanox/x86_64-mlnx_msn4700-r0/plugins/sfplpmget.py [hunk 3]
```diff
@@ -0,0 +1 @@
+../../x86_64-mlnx_msn2700-r0/plugins/sfplpmget.py
```
Comment (liat-grozovik) [inline device/mellanox/x86_64-mlnx_msn4700-r0/plugins/sfplpmget.py]: need to clarify as of the 8 lanes support

File: device/mellanox/x86_64-mlnx_msn4700-r0/sensors.conf [hunk 4]
```diff
@@ -0,0 +1,116 @@
+################################################################################
```
Comment (liat-grozovik) [inline device/mellanox/x86_64-mlnx_msn4700-r0/sensors.conf]: need to clarify to be on the same size as this is not the same system.

File: device/mellanox/x86_64-mlnx_msn4700-r0/sensors.conf [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+# Copyright (c) 2019 Mellanox Technologies
+#
+# Platform specific sensors config for SN3700
```
Comment (liat-grozovik) [inline device/mellanox/x86_64-mlnx_msn4700-r0/sensors.conf]: need to fix

File: device/mellanox/x86_64-mlnx_msn4700_simx-r0/platform_reboot [hunk 6]
```diff
@@ -0,0 +1 @@
+../x86_64-mlnx_msn4700-r0/platform_reboot
```
Comment (liat-grozovik) [inline device/mellanox/x86_64-mlnx_msn4700_simx-r0/platform_reboot]: need to fix to point to a simx reboot file not to real hw system

File: device/mellanox/x86_64-mlnx_msn4700-r0/ACS-MSN4700/port_config.ini (hunk 1 above)
Comment (keboliu) [reply device/mellanox/x86_64-mlnx_msn4700-r0/ACS-MSN4700/port_config.ini]: revised.

File: device/mellanox/x86_64-mlnx_msn4700-r0/plugins/psuutil.py (hunk 2 above)
Comment (keboliu) [reply device/mellanox/x86_64-mlnx_msn4700-r0/plugins/psuutil.py]: verified during the bring up test.

File: device/mellanox/x86_64-mlnx_msn4700-r0/plugins/sfplpmget.py (hunk 3 above)
Comment (keboliu) [reply device/mellanox/x86_64-mlnx_msn4700-r0/plugins/sfplpmget.py]: according to FW, no change for this on Spectrum-3

File: device/mellanox/x86_64-mlnx_msn4700-r0/sensors.conf (hunk 4 above)
Comment (keboliu) [reply device/mellanox/x86_64-mlnx_msn4700-r0/sensors.conf]: revised according to the bring up test result.

File: device/mellanox/x86_64-mlnx_msn4700-r0/sensors.conf (hunk 5 above)
Comment (keboliu) [reply device/mellanox/x86_64-mlnx_msn4700-r0/sensors.conf]: revised

File: device/mellanox/x86_64-mlnx_msn4700_simx-r0/platform_reboot (hunk 6 above)
Comment (keboliu) [reply device/mellanox/x86_64-mlnx_msn4700_simx-r0/platform_reboot]: chnaged

**Comment by liat-grozovik** [comment]
retest vs please

File: device/mellanox/x86_64-mlnx_msn4700-r0/ACS-MSN4700/pg_profile_lookup.ini [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+ 200000    5m      41984     18432    23552     0
+ 400000    5m      48128     19456    28672     0
+   1000    40m     36864     18432    18432     0
+  10000    40m     38912     18432    20480     0
+  25000    40m     41984     18432    23552     0
+  40000    40m     45056     18432    26624     0
+  50000    40m     47104     18432    28672     0
+ 100000    40m     59392     18432    40960     0
+ 200000    40m     81920     18432    63488     0
+ 400000	   40m     128000    19456    108544    0
```
Comment (nazariig) [inline device/mellanox/x86_64-mlnx_msn4700-r0/ACS-MSN4700/pg_profile_lookup.ini]: @keboliu please fix the indentation

Comment (keboliu) [reply device/mellanox/x86_64-mlnx_msn4700-r0/ACS-MSN4700/pg_profile_lookup.ini]: fixed

**Comment by jleveque** [comment]
retest vsimage please

File: device/mellanox/x86_64-mlnx_msn4700-r0/sensors.conf [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+        label power2 "PMIC-4 ASIC 1.8V T2_3 Rail Pwr (out)"
+        label curr1 "PMIC-4 ASIC 0.85V VCORE_T2_3 Rail Curr (out)"
+        label curr2 "PMIC-4 ASIC 1.8V T2_3 Rail Curr (out)"
+    chip "tps53679-i2c-*-6a"
+        label in1 "PMIC-5 PSU 12V Rail (in)"
+        label in2 "PMIC-5 ASIC 0.85V VCORE_T4_5 Rail (out)"
+        label in3 "PMIC-5 ASIC 1.8V T4_5 Rail (out)"
+        label temp1 "PMIC-5 Temp 1"
+        label temp2 "PMIC-5 Temp 2"
+        label power1 "PMIC-5 ASIC 0.85V VCORE_T4_4 Rail Pwr (out)"
```
Comment (nazariig) [inline device/mellanox/x86_64-mlnx_msn4700-r0/sensors.conf]: @keboliu VCORE_T4_5

File: device/mellanox/x86_64-mlnx_msn4700-r0/ACS-MSN4700/buffers_defaults_t1.j2 [hunk 9]
```diff
@@ -0,0 +1,91 @@
+{% set default_cable = '5m' %}
+{% set ingress_lossless_pool_size =  '19283968' %}
+{% set ingress_lossy_pool_size =  '19283968' %}
+{% set egress_lossless_pool_size =  '67108864' %}
```
Comment (nazariig) [inline device/mellanox/x86_64-mlnx_msn4700-r0/ACS-MSN4700/buffers_defaults_t1.j2]: @keboliu we need to recalculate buffers size. You have used incorrect value (64 mb)

File: device/mellanox/x86_64-mlnx_msn4700-r0/ACS-MSN4700/buffers_defaults_t0.j2 [hunk 10]
```diff
@@ -0,0 +1,91 @@
+{% set default_cable = '5m' %}
+{% set ingress_lossless_pool_size =  '30973952' %}
+{% set ingress_lossy_pool_size =  '30973952' %}
+{% set egress_lossless_pool_size =  '67108864' %}
```
Comment (nazariig) [inline device/mellanox/x86_64-mlnx_msn4700-r0/ACS-MSN4700/buffers_defaults_t0.j2]: @keboliu we need to recalculate buffers size. You have used incorrect value (64 mb)

File: device/mellanox/x86_64-mlnx_msn4700-r0/sensors.conf (hunk 8 above)
Comment (keboliu) [reply device/mellanox/x86_64-mlnx_msn4700-r0/sensors.conf]: fixed

File: device/mellanox/x86_64-mlnx_msn4700-r0/ACS-MSN4700/buffers_defaults_t1.j2 (hunk 9 above)
Comment (keboliu) [reply device/mellanox/x86_64-mlnx_msn4700-r0/ACS-MSN4700/buffers_defaults_t1.j2]: fixed

File: device/mellanox/x86_64-mlnx_msn4700-r0/ACS-MSN4700/buffers_defaults_t0.j2 (hunk 10 above)
Comment (keboliu) [reply device/mellanox/x86_64-mlnx_msn4700-r0/ACS-MSN4700/buffers_defaults_t0.j2]: fixed

**Comment by liat-grozovik** [comment]
retest vsimage please

File: device/mellanox/x86_64-mlnx_msn4700-r0/ACS-MSN4700/buffers_defaults_t1.j2 [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+            "profile" : "[BUFFER_PROFILE|egress_lossless_profile]"
+        },
+        "{{ port_names }}|0-1": {
+            "profile" : "[BUFFER_PROFILE|q_lossy_profile]"
+        },
+        "{{ port_names }}|5": {
+            "profile" : "[BUFFER_PROFILE|q_lossy_profile]"
+        }
+    }
+{%- endmacro %}
```
Comment (stephenxs) [inline device/mellanox/x86_64-mlnx_msn4700-r0/ACS-MSN4700/buffers_defaults_t1.j2]: The key for BUFFER_QUEUE table is different from what it is for other switches. 
In the 4700 the key is like
```
"BUFFER_QUEUE|Ethernet0,Ethernet4,Ethernet8,Ethernet12,Ethernet16,Ethernet20,Ethernet24,Ethernet28,Ethernet32,Ethernet36,Ethernet40,Ethernet44,Ethernet48,Ethernet52,Ethernet56,Ethernet60,Ethernet64,Ethernet68,Ethernet72,Ethernet76,Ethernet80,Ethernet84,Ethernet88,Ethernet92,Ethernet96,Ethernet100,Ethernet104,Ethernet108,Ethernet112,Ethernet116,Ethernet120,Ethernet124|3-4"
```
while in 2700, 3700 there is a key for each port, like `BUFFER_QUEUE|Ethernet4|3-4`
This difference may break qos test.

Comment (keboliu) [reply device/mellanox/x86_64-mlnx_msn4700-r0/ACS-MSN4700/buffers_defaults_t1.j2]: @stephenxs updated, please review.

Comment (stephenxs) [reply device/mellanox/x86_64-mlnx_msn4700-r0/ACS-MSN4700/buffers_defaults_t1.j2]: @keboliu It's correct now.

**Comment by keboliu** [comment]
retest vsimage please

**Comment by keboliu** [comment]
retest vsimage please

**Comment by keboliu** [comment]
retest this please

**Comment by keboliu** [comment]
retest this, please

**Comment by keboliu** [comment]
retest this please

**Comment by keboliu** [comment]
retest vsimage please

**Comment by keboliu** [comment]
retest vsimage please

File: device/mellanox/x86_64-mlnx_msn4700-r0/ACS-MSN4700/pg_profile_lookup.ini [hunk 12]
```diff
@@ -0,0 +1,25 @@
+# speed  cable   size    xon     xoff   threshold
```
Comment (nazariig) [inline device/mellanox/x86_64-mlnx_msn4700-r0/ACS-MSN4700/pg_profile_lookup.ini]: @keboliu please fix alignment

Comment (nazariig) [inline device/mellanox/x86_64-mlnx_msn4700-r0/ACS-MSN4700/pg_profile_lookup.ini]:
```suggestion
# speed  cable   size    xon     xoff     threshold
1000     5m      32768   18432   14336    0
```

**Comment by keboliu** [comment]
retest this please

**Comment by keboliu** [comment]
retest this please

**Comment by keboliu** [comment]
retest vsimage please

**Comment by keboliu** [comment]
retest vsimage please

File: platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py [hunk 13]
```diff
@@ -49,7 +49,7 @@
 
 # magic code defnition for port number, qsfp port position of each hwsku
 # port_position_tuple = (PORT_START, QSFP_PORT_START, PORT_END, PORT_IN_BLOCK, EEPROM_OFFSET)
-hwsku_dict_port = {'ACS-MSN2010': 3, 'ACS-MSN2100': 1, 'ACS-MSN2410': 2, 'ACS-MSN2700': 0, 'Mellanox-SN2700': 0, 'Mellanox-SN2700-D48C8': 0, 'LS-SN2700':0, 'ACS-MSN2740': 0, 'ACS-MSN3700': 0, 'ACS-MSN3700C': 0, 'ACS-MSN3800': 4, 'Mellanox-SN3800-D112C8': 4}
+hwsku_dict_port = {'ACS-MSN2010': 3, 'ACS-MSN2100': 1, 'ACS-MSN2410': 2, 'ACS-MSN2700': 0, 'Mellanox-SN2700': 0, 'Mellanox-SN2700-D48C8': 0, 'LS-SN2700':0, 'ACS-MSN2740': 0, 'ACS-MSN3700': 0, 'ACS-MSN3700C': 0, 'ACS-MSN3800': 4, 'Mellanox-SN3800-D112C8': 4, 'ACS-MSN4700': 0}
```
Comment (keboliu) [inline platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py]: here you need to add the new SKU name "ACS-MSN4600" or "ACS-MSN4600C", need you to confirm the name.

File: platform/mellanox/mlnx-platform-api/sonic_platform/thermal.py [hunk 14]
```diff
@@ -106,7 +106,7 @@
     THERMAL_API_GET_HIGH_THRESHOLD
 ]
 
-hwsku_dict_thermal = {'ACS-MSN2700': 0, 'LS-SN2700':0, 'ACS-MSN2740': 3, 'ACS-MSN2100': 1, 'ACS-MSN2410': 2, 'ACS-MSN2010': 4, 'ACS-MSN3700': 5, 'ACS-MSN3700C': 6, 'Mellanox-SN2700': 0, 'Mellanox-SN2700-D48C8': 0, 'ACS-MSN3800': 7, 'Mellanox-SN3800-D112C8': 7}
+hwsku_dict_thermal = {'ACS-MSN2700': 0, 'LS-SN2700':0, 'ACS-MSN2740': 3, 'ACS-MSN2100': 1, 'ACS-MSN2410': 2, 'ACS-MSN2010': 4, 'ACS-MSN3700': 5, 'ACS-MSN3700C': 6, 'Mellanox-SN2700': 0, 'Mellanox-SN2700-D48C8': 0, 'ACS-MSN3800': 7, 'Mellanox-SN3800-D112C8': 7, 'ACS-MSN4700': 8}
```
Comment (keboliu) [inline platform/mellanox/mlnx-platform-api/sonic_platform/thermal.py]: same as above

File: platform/mellanox/mlnx-platform-api/sonic_platform/thermal.py [hunk 15]
```diff
@@ -231,6 +231,22 @@
...
+        THERMAL_DEV_CATEGORY_AMBIENT:(0,
+            [
+                THERMAL_DEV_ASIC_AMBIENT,
+                THERMAL_DEV_COMEX_AMBIENT,
+                THERMAL_DEV_PORT_AMBIENT,
+                THERMAL_DEV_FAN_AMBIENT
+            ]
+        )
+    }
 ]
```
Comment (keboliu) [inline platform/mellanox/mlnx-platform-api/sonic_platform/thermal.py]: need add specific data for the new SKU.

**Comment by keboliu** [comment]
retest this please

**Comment by liat-grozovik** [comment]
retest vsimage please

**Comment by keboliu** [comment]
retest vsimage please

**Comment by keboliu** [comment]
retest vsimage please

**Comment by keboliu** [comment]
retest vsimage please

**Comment by liat-grozovik** [comment]
@jleveque can you please check if your comments were addressed and you can approve it?
we would like to have this new system part of 201911 the sooner the better.

**Pull request merged**